src/transformers/utils/import_utils.py

```diff
@@ -1509,19 +1515,73 @@ class _LazyModule(ModuleType):
     # https://github.com/optuna/optuna/blob/master/optuna/integration/__init__.py
     def __init__(self, name, module_file, import_structure, module_spec=None, extra_objects=None):
         super().__init__(name)
-        self._modules = set(import_structure.keys())
-        self._class_to_module = {}
-        for key, values in import_structure.items():
-            for value in values:
-                self._class_to_module[value] = key
-        # Needed for autocompletion in an IDE
-        self.__all__ = list(import_structure.keys()) + list(chain(*import_structure.values()))
-        self.__file__ = module_file
-        self.__spec__ = module_spec
-        self.__path__ = [os.path.dirname(module_file)]
-        self._objects = {} if extra_objects is None else extra_objects
-        self._name = name
-        self._import_structure = import_structure
+
+        if any(isinstance(key, frozenset) for key in import_structure.keys()):
+            PER_BACKEND_SPLIT = True
+        else:
+            PER_BACKEND_SPLIT = False
+
+        self._object_missing_backend = {}
+
+        if PER_BACKEND_SPLIT:
+            self._modules = set()
+            self._class_to_module = {}
+            self.__all__ = []
+
+            _import_structure = {}
+
+            for backends, item in import_structure.items():
+                lacking_backends = []
+                for backend in backends:
+                    if backend not in BACKENDS_MAPPING:
+                        raise ValueError(
+                            f"Error: the following backend: '{backend}' was specified around object {item} but isn't specified in the backends mapping."
+                        )
+                    callable, error = BACKENDS_MAPPING[backend]
+                    if not callable():
+                        lacking_backends.append(backend)
+
+                self._modules.union(set(item.keys()))
+                for key, values in item.items():
+                    for value in values:
+                        self._class_to_module[value] = key
+
+                    if key not in _import_structure:
+                        _import_structure[key] = values
+                    else:
+                        _import_structure[key].extend(values)
+
+                # Needed for autocompletion in an IDE
+                self.__all__.extend(list(item.keys()) + list(chain(*item.values())))
+
+                if len(lacking_backends):
+                    for module, objects in item.items():
+                        for obj in objects:
```

Comment:
same here

Reply:
while this could be done with the firs toop line 1545 no?

Reply:
Applied!